src/server/ua_server.c

```diff
@@ -807,6 +921,19 @@ UA_Server_run_shutdown(UA_Server *server) {
     UA_Server_removeCallback(server, server->houseKeepingCallbackId);
     server->houseKeepingCallbackId = 0;
 
+    /* Mark all reverse connects as destroying */
+    reverse_connect_context *rev = NULL;
+    SLIST_FOREACH(rev, &server->reverseConnects, next) {
+        rev->destruction = true;
+        if (rev->currentConnection.connectionId) {
+            rev->currentConnection.connectionManager->closeConnection(rev->currentConnection.connectionManager,
+                                                                      rev->currentConnection.connectionId);
+        } else if (rev->stateCallback) {
+            rev->stateCallback(server, rev->handle, UA_SECURECHANNELSTATE_CLOSED,
```

Comment:
You can use `setReverseConnectState` in every case (not just for else-if) to signal state changes.